plugins/plugins-api/src/main/java/org/codice/compliance/saml/plugin/IdpResponse.java

```diff
@@ -0,0 +1,52 @@
+package org.codice.compliance.saml.plugin;
+
+import org.w3c.dom.Node;
+
+public abstract class IdpResponse {
+  protected IdpResponse() {}
+
+  protected int httpStatusCode;
+  protected String samlResponse;
+  protected String relayState;
+
+  protected boolean isRelayStateGiven;
+
+  private String decodedSamlResponse;
+  private Node responseDom;
+
+  public int getHttpStatusCode() {
+    return httpStatusCode;
+  }
+
+  public String getSamlResponse() {
+    return samlResponse;
+  }
+
+  public String getRelayState() {
+    return relayState;
+  }
+
+  public boolean isRelayStateGiven() {
+    return isRelayStateGiven;
+  }
+
+  public void setRelayStateGiven(boolean relayStateGiven) {
+    this.isRelayStateGiven = relayStateGiven;
+  }
+
+  public String getDecodedSamlResponse() {
+    return decodedSamlResponse;
+  }
+
+  public void setDecodedSamlResponse(String decodedSamlResponse) {
+    this.decodedSamlResponse = decodedSamlResponse;
+  }
+
+  public Node getResponseDom() {
+    return responseDom;
+  }
+
+  public void setResponseDom(Node responseDom) {
+    this.responseDom = responseDom;
```

Comment:
These setters are public on purpose. Some of the values are set by our code after the object is built. For instance: `decodedSamlResponse` is set when we decode the saml response in verification, not during the object build time.

Reply:
That's fair. Though that makes me think that we're getting a little confusing here because we're technically storing information for multiple responses at that point, right? I think we should make that fact clear in the object. By naming methods/fields and possibly talking about it in Javadoc.

Reply:
>we're technically storing information for multiple responses at that point, right?
It is the same exact samlResponse that is stored in the built `samlResponse` property, just decoded.

But I always agree with more commentation. I'll try to push something up.

Reply:
👍 Right. It's just various representations of the response. Duh 😄

Reply:
❓ Is there value to holding these other representations at this level, or will they be moved to a decorator at some point?